src/if_cscope.c

```diff
@@ -315,46 +315,46 @@ ex_cstag(exarg_T *eap)
     case 0 :
 	if (cs_check_for_connections())
 	{
-	    ret = cs_find_common("g", (char *)(eap->arg), eap->forceit, FALSE,
-						       FALSE, *eap->cmdlinep);
-	    if (ret == FALSE)
+	    ret = cs_find_common("g", (char *)(eap->arg), eap->forceit, false,
+						       false, *eap->cmdlinep);
+	    if (ret == false)
```

Comment:
Avoid writing conditions like `if (ret == false)`.
`ret` is already a boolean, so you can simply write `if (!ret)` instead.
It's shorter, more idiomatic, and easier to read.

```suggestion
	    if (!ret)
```

Reply:
Thanks for your feedback.
I think it would be better to keep this PR about migrating only from TRUE/FALSE to true/false and have a separate PR for style.
Besides, I personally find `if (ret == false)` more legible than `if (!ret)` ¯\\_(ツ)_/¯

Reply:
> Besides, I personally find `if (ret == false)` more legible than `if (!ret)

So why did you convert it to bool type? It's meaningless unless you eliminate that (comparison with `true` and `false`).

---

@chrisbra 
That aside, I think it would be better to convert TRUE and FALSE to true and false all at once for the entire project.
(In this state, there should be no build errors or warnings.)

After that, it would make sense to use `clang-tidy` to find places where `true` and `false` are assigned to `int` and convert them from `int` to `bool`.

Example `.clang-tidy` config (untested):
```yaml
Checks: >
  -*,
  readability-implicit-bool-conversion

CheckOptions:
  - key: readability-implicit-bool-conversion.AllowIntegerConditions
    value: 'false'
  - key: readability-implicit-bool-conversion.WarnOnIntegerAssignment
    value: 'true'
  - key: readability-implicit-bool-conversion.WarnOnBooleanToIntegerAssignment
    value: 'true'
WarningsAsErrors: '*'
```

Example execution (untested):
```bash
$ clang-tidy test.c -- -std=c99
```

Reply:
It also seems possible to detect `if (x == true/false)`.
(untested)
```yaml
Checks: >
  -*,
  readability-implicit-bool-conversion,
  readability-simplify-boolean-expr

CheckOptions:
  # 1. Disallow assigning bool literals (true/false) to int
  - key: readability-implicit-bool-conversion.AllowIntegerConditions
    value: 'false'
  - key: readability-implicit-bool-conversion.WarnOnIntegerAssignment
    value: 'true'
  - key: readability-implicit-bool-conversion.WarnOnBooleanToIntegerAssignment
    value: 'true'

  # 2. Disallow if (x == true/false)
  - key: readability-simplify-boolean-expr.ChainedConditionalReturn
    value: 'true'
  - key: readability-simplify-boolean-expr.SimplifyConditionals
    value: 'true'

WarningsAsErrors: '*'
```

Reply:
> That aside, I think it would be better to convert TRUE and FALSE to true and false all at once for the entire project.

The main benefit is not about TRUE/FALSE but rather about using `bool` instead of `int` when possible. And this cannot be automated, as some functions that are currently using TRUE/FALSE still need to have `int` as return type. One example in this file is `cs_find`.

Reply:
First of all, I didn't say "automate."

My suggestions are:
1. Replace TRUE/FALSE with true/false all at once (with one PR).
2. Use `clang_tidy` to find where `int` should be changed to `bool` and make the change manually (with multiple PRs).

> ...still need to have int as return type. One example in this file is cs_find.

??. Why can't it be `bool`?
The return type of `cs_find()` could be `bool`.
We can change the return type of the member `func` of `cscmd_T` to `bool`. (The return value of `func` is not referenced in the first place.)

Reply:
The return value of `func` cannot be changed to `bool`, for example `cs_add` returns `CSCOPE_SUCCESS` or `CSCOPE_FAILURE`, that’s either `0` or `-1`.

Reply:
If you do not address the issues, the merge will not take place.